File: src/c4/inter/Logical.java
package inter;

import lexer.*;

/*
 * Logical ist eine Unterklasse von Expr und beschreibt logische Ausdrücke. 
 * In den Instanzenvariablen expr1 und expr2 werden die beiden Operanden 
 * abgelegt
 */

public abstract class Logical extends Expr {
	Expr expr1, expr2;

	Logical(Token tok, Expr x1, Expr x2) {
		super(tok, null);
		expr1 = x1;
		expr2 = x2;
	}

	public Expr getExpr1() {
		return expr1;
	}

	public Expr getExpr2() {
		return expr2;
	}

	public void setExpr1(Expr expr1) {
		this.expr1 = expr1;
	}

	public void setExpr2(Expr expr2) {
		this.expr2 = expr2;
	}

}

File: src/c4/inter/Logical.rs
use super::{Expr, ExprData, Token};

/// State shared by `And`, `Or`, `Not`, and `Rel`.
#[derive(Clone, Debug, PartialEq)]
pub struct LogicalData {
    pub(crate) expr: ExprData,
    expr1: Expr,
    expr2: Expr,
}

impl LogicalData {
    pub fn new(token: Token, expr1: Expr, expr2: Expr) -> Self {
        Self {
            expr: ExprData::new(token, None),
            expr1,
            expr2,
        }
    }

    pub fn get_expr1(&self) -> &Expr {
        &self.expr1
    }
    pub fn get_expr2(&self) -> &Expr {
        &self.expr2
    }
    pub fn set_expr1(&mut self, expr: Expr) {
        self.expr1 = expr;
    }
    pub fn set_expr2(&mut self, expr: Expr) {
        self.expr2 = expr;
    }
}
